GulajavaMinistudio/yarn
# Update upstream

* nohoist impl check point

- nohoist implementation
- 'why' command fixes
- 'add' command fixes
- tests and test fixtures

see [RFC #86](https://github.com/yarnpkg/rfcs/pull/86) for detail

* fix not adding devDependencies

* add nohoist flag and eligibility check

1. added a new flags 'workspaces-nohoist-experimental' to disable nohoist.
2. added eligibility validation in Config.getWorkspaces, violation will be reported and config be ignored.
3. update test fixtures to add private flag for nohoist tests

* revert path separator to '#' for display

* pass through private flag in root manifest

* fix lint error

* addressing @bestander review comments on 1/8

* fix merge conflict

* fix merge lint issues

* address @arcanis comments

* update snapshot after merge

* one more snapshot update



**Summary**





**Test plan**
